### PR TITLE
Add YAML spec handling

### PR DESCRIPTION
**Pull request by davidcochrum**
Allow for the spec file referenced in the route definition to be `.json`, `.yaml`, or `.yml`

### CODE REVIEW

File: src/Json/SchemaLoader.php [hunk 1]
```diff
@@ -61,7 +62,12 @@ public function load(string $file): stdClass
         $locatedFile = $this->fileLocator->locate($file);
 
         if (isset($this->schemas[$locatedFile]) === false) {
-            $schema = $this->dereferencer->dereference('file://'.$locatedFile);
+            if (in_array(pathinfo($locatedFile, PATHINFO_EXTENSION), array('yml', 'yaml'))) {
+                $dereference = (object) Yaml::parseFile($locatedFile);
+            } else {
+                $dereference = "file://{$locatedFile}";
```
Comment (gavinkalika) [inline src/Json/SchemaLoader.php]: perhaps a check to determine that this file has json extension....sorta like you did with the yaml check

if not then throw an exception as your default clause.

Comment (davidcochrum) [reply src/Json/SchemaLoader.php]: If not a `.yaml` or `.yml` extension, handling falls back to the same as it was ... if it's not JSON, an error should be generated

Comment (davidcochrum) [reply src/Json/SchemaLoader.php]:
```shell
$ php bin/console debug:router

In FileLoader.php line 168:
                                                                                                                                                                                                                                      
  Invalid JSON: Syntax error in ../src/Resources/api/v1/foo.txt (which is being imported from "/path/to/routes.yaml"). Make sure there is a loader supporting the "openapi" type.                                                                                                                                                                                                                           
                                                                                                                                                                                                                                      
In JsonDecoder.php line 46:
                              
  Invalid JSON: Syntax error 
```

File: src/Json/SchemaLoader.php [hunk 2]
```diff
@@ -61,7 +62,12 @@ public function load(string $file): stdClass
         $locatedFile = $this->fileLocator->locate($file);
 
         if (isset($this->schemas[$locatedFile]) === false) {
-            $schema = $this->dereferencer->dereference('file://'.$locatedFile);
+            if (in_array(pathinfo($locatedFile, PATHINFO_EXTENSION), array('yml', 'yaml'))) {
+                $dereference = (object) Yaml::parseFile($locatedFile);
```
Comment (niels-nijens) [inline src/Json/SchemaLoader.php]: The Yaml parser can also return an object/stdClass as output:
```suggestion
                $dereference = Yaml::parseFile($locatedFile, Yaml::PARSE_OBJECT_FOR_MAP);
```

File: src/Json/SchemaLoader.php [hunk 3]
```diff
@@ -54,14 +56,30 @@ public function __construct(FileLocatorInterface $fileLocator, DereferencerInter
...
+                case 'yaml':
+                    $dereference = Yaml::parseFile($locatedFile, Yaml::PARSE_OBJECT_FOR_MAP);
+                    break;
+
+                case 'json':
+                    $dereference = "file://{$locatedFile}";
+                    break;
+
+                default:
+                    throw new FileLoaderLoadException($locatedFile);
```
Comment (davidcochrum) [inline src/Json/SchemaLoader.php]: converted to a `switch/case` block which inspects the extension to take the appropriate action or throw an exception ... is `.json5` supported?

Comment (niels-nijens) [reply src/Json/SchemaLoader.php]: Looks good. 👍 PHP doesn't have native support for JSON5 I believe.

**Comment by niels-nijens** [comment]
Thanks for this contribution!

**Comment by davidcochrum** [comment]
Will you be cutting a release for these changes? I can drop my fork when you do

**Comment by niels-nijens** [comment]
I've just released 0.3.0.

**Pull request merged**